fronts-client/src/util/extractAtomId.ts

```diff
@@ -0,0 +1,136 @@
+import urlConstants from '../constants/url';
+import type { Atom, AtomAsset, ImageAsset, Platform } from '../types/Capi';
+
+export const stripQueryParams = (value: string) => {
+	const parts: string[] = value.split('?');
+	return parts[0];
+};
+
+const attemptResourceExtract = (
+	fullPath: string,
+	baseUrl: string,
+	resourcePath: string,
+): string | undefined => {
+	if (fullPath.includes(baseUrl) && fullPath.includes(resourcePath)) {
+		const parts: string[] = fullPath.split(resourcePath);
+		if (parts.length > 1) return parts[1];
+	}
+
+	return undefined;
+};
+
+const extractAtomId = (videoUri: string | undefined): string => {
+	/*
+	 * This method works for both PROD and CODE paths, irrespective of the stage of the Fronts Tool.
+	 *
+	 * We could choose to pass a stage parameter, but I think it's better to be flexible, given
+	 * CODE Fronts Tool can be pointed to PROD or CODE CAPI.
+	 */
+	if (videoUri === undefined) return '';
+
+	const cleanVideoUri = stripQueryParams(videoUri);
+
+	const atomMakerAttempt = attemptResourceExtract(
+		cleanVideoUri,
+		urlConstants.video.videoBaseUrl,
+		urlConstants.video.mediaAtomMakerPath,
+	);
+	if (atomMakerAttempt !== undefined)
+		return `${urlConstants.video.capiAtomPath}${atomMakerAttempt}`;
+
+	const capiAttempt = attemptResourceExtract(
+		cleanVideoUri,
+		urlConstants.base.capi,
+		urlConstants.video.capiAtomPath,
+	);
+	if (capiAttempt !== undefined)
+		return `${urlConstants.video.capiAtomPath}${capiAttempt}`;
+
+	return '';
+};
+
+const extractAssetId = (atom: Atom): string | undefined => {
+	const assets: AtomAsset[] = atom.data?.media?.assets;
+	if (
+		assets === undefined ||
+		assets.length === 0 ||
+		assets[0] === undefined ||
+		assets[0].id === undefined
+	) {
+		throw new Error(`No assets found for atom ${atom.id}`);
+	} else {
+		return assets[0].id;
+	}
+};
```

Comment:
Non-blocking suggestion - we could simplify this with optional chaining eg
```suggestion
  const assetId = atom.data?.media?.assets?.[0]?.id;
  if (!assetId) {
    throw new Error(`No assets found for atom ${atom.id}`);
  }
  return assetId;
};
```